src/App.tsx

```diff
@@ -14,11 +15,106 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  status: string;
+  selected: string;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    // eslint-disable-next-line react/no-unused-state
+    status: 'No goods selected',
+    selected: 'selected',
+    // eslint-disable-next-line react/no-unused-state
+  };
+
+  selectedGoods = ['Jam'];
+
+  componentDidMount() {
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    this.setState({ status: `${this.selectedGoods} is selected` });
+  }
```

Comment:
you can remove componentDidMount and add this message to state `${this.selectedGoods} is selected`